### PR TITLE
JAVA-8292 : Split or move spring-data-jpa-enterprise module

### PR DESCRIPTION
**Pull request by anuragkumawat**

### CODE REVIEW

File: persistence-modules/spring-data-jpa-enterprise-2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
```
Comment (kwoyke) [inline persistence-modules/spring-data-jpa-enterprise-2/pom.xml:26]: jdbs and cache dependencies are not needed. Please remove them.

File: persistence-modules/spring-data-jpa-enterprise-2/pom.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            <artifactId>h2</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>${guava.version}</version>
+        </dependency>
+    </dependencies>
+
+    <build>
```
Comment (kwoyke) [inline persistence-modules/spring-data-jpa-enterprise-2/pom.xml:44]: This build section is redundant. Remove please.

Comment (anuragkumawat) [reply persistence-modules/spring-data-jpa-enterprise-2/pom.xml]: Removed this build section

File: persistence-modules/spring-data-jpa-enterprise-2/pom.xml (hunk 1 above)
Comment (anuragkumawat) [reply persistence-modules/spring-data-jpa-enterprise-2/pom.xml]: JDBC and Cache Dependencies are removed.

**Pull request merged**